### ISSUE TITLE
Add "Single Use" variants of all the tools for use in AE autocrafting

### ISSUE THREAD

**Issue by RubiksImplosion** [GTNewHorizons/GT-New-Horizons-Modpack#14102]
### Your GTNH Discord Username

rubiksimplosion

### Your Pack Version

Latest Nightly (2.3.5 Base)

### Your Proposal

Add a "Single Use" version of the Wrench, Hammer, File, Screwdriver, Wire Cutter, Saw, Crowbar, Mortar, and Soft Mallet. Their cost should be completely negligible for late-game (iron/steel/aluminum/etc...)

### Your Goal

There are hundreds of recipes that must be patterned using a tool of some sort. Doing this currently requires patterning anything that needs a tool with "Use Substitutions: Yes" which results in hundreds of tools with varying levels of durability ending up in the ME System. It also results in any crafts that use tools taking a significant amount of time to calculate.

This can already be done by using one of the 0 durability tools that exist (Sunnarium/Magnetic Samarium for tools and Livingwood for Soft Mallet are the cheapest ways), but is an excessively expensive solution. Having a dedicated cheap item that is intended to be used would make this future-proof and much nicer for players.

This is also a significantly easier solution to implement than to find and write assembler recipes for the hundreds of different tool-only items that currently exist.

### Your Vision

Adding this would decrease lag from AE recalculations, make patterning things with tools much more convenient, and decrease AE clutter.

### Final Checklist

- [X] I have searched this issue tracker and there is nothing similar already. Posting on a closed issue saying *I like this feature please reconsider adding it* will prompt us to investigate and reopen it once we confirm your report.
- [X] I believe there is nothing similar in the pack already, or the existing solution isn't good enough.
- [X] I understand this change request may not attract enough attention and thus not be implemented.
- [X] I understand this change request may be rejected due to other community members thinking it's inappropriate.
- [X] I believe this feature would make the pack better.

**Comment by Glease** [comment]
I have code ready for this a few month ago. it never got pushed because I'm not sure about the balance. Given we are currently in a feature freeze this will have to wait till next stable is released, but we can talk about the price first.

**Comment by cainiaojun** [comment]
The Multi Molecular Assembler behaves badly when got related to tool-used craftings - it would void those tool! Hope the disposable tools could fix this problem. 

The AE network calculates EXTREME SLOWLY when being involved in tool-used recipes. If we uses disposable tools, it would be much better since there would be no need for AE to calculate the durability of tools. 

And the disposable tools should be stackable. Otherwise it would take up a lot of ME storage room. I think the mechanism of Mixed Metal Ingot is good. We can craft 1 ingot with Bronze plate, Tin plate and Iron plate at a time, or craft 63 ingot with Neutonium Plate, Bedrockium Plate and Awakened Draconium Plate at a time. The disposable tools could obey the same rules. For instance: 
 - 1 disposible hammer, crafted by 3 iron ingot and 1 stick
 - 16 disposible hammer, crafted by 3 titanium ingot and 1 steel stick
 - 64 disposible hammer, crafted by 3 neutonium ingot and 1 tungsteen steel stick.
Players are free to choose what recipe they would like to use at all. The beginning recipe is cheapest while the last recipe enable players to craft a huge amount of tools in a short period.

**Comment by chochem** [comment]
this is not necessarily a reason against this, but out current approach has been to add a bunch of assembler recipes. There shouldn't be much left where you need tools in ae2. At least not in the bigger nested crafts.

**Comment by RubiksImplosion** [comment]
> this is not necessarily a reason against this, but out current approach has been to add a bunch of assembler recipes. There shouldn't be much left where you need tools in ae2. At least not in the bigger nested crafts.

Adding explicit assembler recipes for all the hundreds of tool stuff is one of the lowest-priority tasks. Adding single use variants would be a very easy to implement band-aid fix that doesn't harm anything and only helps players, since adding assembler recipes for everything could take literal years.

**Comment by desagas** [comment]
I think it would be better to add another button beside use subs for tools, so that it only allows for tools. If this is possible.

**Comment by RubiksImplosion** [comment]
> I think it would be better to add another button beside use subs for tools, so that it only allows for tools. If this is possible.

This doesn't solve the problem of AE clutter due to tools. Any time a tool is used it gets put back into the system with some durability taken. This results in the AE system getting cluttered with dozens of tools of various durability levels.

**Comment by chochem** [comment]
> > this is not necessarily a reason against this, but out current approach has been to add a bunch of assembler recipes. There shouldn't be much left where you need tools in ae2. At least not in the bigger nested crafts.
> 
> Adding explicit assembler recipes for all the hundreds of tool stuff is one of the lowest-priority tasks. Adding single use variants would be a very easy to implement band-aid fix that doesn't harm anything and only helps players, since adding assembler recipes for everything could take literal years.

hardly. most relevant ones are done. Doesnt really matter for one-time crafts. only stuff that is either massproduced or in more complex crafting chains.

**Comment by desagas** [comment]
> > I think it would be better to add another button beside use subs for tools, so that it only allows for tools. If this is possible.
> 
> This doesn't solve the problem of AE clutter due to tools. Any time a tool is used it gets put back into the system with some durability taken. This results in the AE system getting cluttered with dozens of tools of various durability levels.

No, if it is accurately pulling tools, you would only ever have 1 of a single tool type in AE at a time. Unless you want multiple of them crafting at a time. Also, AE can also craft directly inside of molecular assemblers with 1 pattern per assembler, which means that, you could set up a system to always keep the tools outside of ME< and inject into the molecular assembler only when a craft is happening. This feature has existed since AE2 was created.

**Comment by Solotov0** [comment]
What about a tool that just doesn't use durability?  Could be made to have bad in-hand stats to encourage crafting only.  If we're saying material cost is negligible by the time this is an issue, that could be a goal to shoot for, make the 'unbreakable' GT tool set

**Comment by boubou19** [comment]
as chochem said, we want to remove the tools in craft at endgame by using assembler version of the crafts. Single use tools are just a duct tape, proper solution is to provide an alternative recipe.

**Comment by RubiksImplosion** [comment]
This still doesn't address the issue of thaumcraft arcane assembler recipes that require tools not having any alternate though?
